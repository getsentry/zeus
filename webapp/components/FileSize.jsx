import React, {Component} from 'react';
import PropTypes from 'prop-types';

export default class FileSize extends Component {
  static propTypes = {
    value: PropTypes.number.isRequired
  };

  getSize() {
    let {value} = this.props;
    if (value === 0) return '0 B';
    if (!value) return null;

    let i = Math.floor(Math.log(value) / Math.log(1024));
    return (
      (value / Math.pow(1024, i)).toFixed(2) * 1 + ' ' + ['B', 'kB', 'MB', 'GB', 'TB'][i]
    );
  }

  render() {
    return <span>{this.getSize()}</span>;
  }
}
