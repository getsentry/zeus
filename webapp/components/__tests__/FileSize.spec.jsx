import React from 'react';
import {shallow} from 'enzyme';

import FileSize from '../FileSize';

describe('FileSize', () => {
  it('renders bytes', () => {
    const tree = shallow(<FileSize value={413} />);
    expect(tree).toMatchSnapshot();
  });
});
