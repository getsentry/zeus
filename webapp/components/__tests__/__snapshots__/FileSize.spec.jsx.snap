// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileSize renders bytes 1`] = `
<span>
  413 B
</span>
`;
